_posts/2016-06-08-component-relative-paths-in-angular-2.md

```diff
@@ -0,0 +1,305 @@
+---
+layout:     post
+title:      "Component-Relative Paths in Angular 2"
+
+date: 2016-06-08
+imageUrl: '/images/banner/opaque-tokens-in-angular-2.jpeg'
+
+summary: "Creating components in Angular 2 has been improved in many ways. But developers should be careful when using external HTML and CSS files. Component-relative paths are not immediatley intuitive. But with a brief review of the relative issues and constraints, developers can more easily create maintainable, reusable, portable components."
+
+categories:
+  - angular
+
+tags:
+  - angular2
+  - components
+  - relative paths
+
+topic: components
+
+author: thomasburleson
+---
+
+Easily creating components is the most-loved feature of Angular 2. By now you should be familiar with using the @Component annotations to create components. And I bet you are also familiar with the required metadata information such as `selector` and `template`.
+
+If you happen to be lucky, your components and their HTML/CSS load without any problems. But I am willing to bet that you will or have already encountered the dreaded, frustrating 404 errors where the template HTML or styles (CSS) cannot be found! Let's talk about why that happens and see how we can solve that problem in a way that is flexible and portable. 
```

Comment:
I like to avoid speaking from a self perspective "I" where possible.

"But I am willing to bet..." maybe rather something like "... If not, you have already encountered the dreaded...."

Perfect use of "we" in the last sentence!